docs/developer-interface.rst

```diff
@@ -78,6 +78,11 @@ Dataset
 .. autoclass:: tamr_unify_client.dataset.resource.Dataset
   :members:
 
+Dataset URI
```

Comment:
Currently the docs sections are in alphabetical order, up for discussion whether they should be